### DISCUSSION TITLE
CameraPreview

### DISCUSSION THREAD

**Discussion by zxopink** [CommunityToolkit/Maui#648] [New Feature Discussions]
Capture frame and render to view, something like this:
![image](https://user-images.githubusercontent.com/30203588/191580968-2d2f9bd2-970e-4185-a01e-e74ed941a0cf.png)
with the option to hook on new frames.

**Comment by zxopink** [comment]
I have made a View element with real-time camera frame captures using the `Android.Hardware.Camera2` API but it's only available for the Android platform.

**Comment by pictos** [comment]
Hello there, we already have a spec for CameraView and started the implementation, you can see it here
- https://github.com/CommunityToolkit/Maui/issues/259

**Reply by zxopink** [reply]
Oh neat, is there any way for me to contribute and help with the implementation?

**Reply by pictos** [reply]
@zxopink you can try what exists right now and provide feedback. Community PRs will be accepted after we release the first version.

**Reply by zxopink** [reply]
I tested it a bit and it works great, although I did get confused with naming.
CameraView has a `Shutter` function which takes a picture using the device's camera.
but it also 'has' a `CaptureAsync` function, an extension method to `View` (in `ViewExtensions`) that takes a screenshot of the view.

Couldn't figure out why my handlers weren't called until I noticed `CaptureAsync` is a completely different method.
Overriding `CaptureAsync` with `Shutter` could spare headache for some developers.

Overall it's really easy to setup, got it to work within minutes.
although an option to use the flashlight, change back/front camera and changing the `CameraView` width/height features would be also very nice

EDIT: by width and height I mean ratio, I see it's only 1:1

**Comment by CliffAgius** [comment]
Closed as answered see issue #106
